runtime/lua/vim/lsp/util.lua

```diff
@@ -772,7 +772,7 @@ function M.locations_to_items(locations)
             filename = fname,
             lnum = row + 1,
             col = col + 1;
-            text = temp.msg;
+            text = temp.msg or line;
```

Comment:
I think both Location and LocationLink don't have message field.
So can we delete codes about msg?

Reply:
I think the `message` field is related to having used `set_loclist` to populate diagnostics: 

https://github.com/neovim/neovim/blob/ff1730373c6139db14b8f2f9b24d4ccd7fcfb01d/runtime/lua/vim/lsp/callbacks.lua#L35

I also use an override like:

```lua
local default_diagnostics_callback = lsp.callbacks['textDocument/publishDiagnostics']
local function diagnostics_callback(err, method, result, client_id)
    default_diagnostics_callback(err, method, result, client_id)
    if result and result.diagnostics and result.uri then
        local current_buf = api.nvim_get_current_buf()
        if vim.uri_to_bufnr(result.uri) == current_buf and myutil.exists(vim.uri_to_fname(result.uri)) then
            for _, v in ipairs(result.diagnostics) do
                v.uri = v.uri or result.uri
            end
            util.set_loclist(result.diagnostics)
        end
    end
end
```

But I agree that a method called `locations_to_items` should not deal with turning diagnostics into items as well. But maybe removing the message can be a follow up in a separate PR?

It probably makes sense to provide some kind of alternative for people who want to have the diagnostics in the location list.

Reply:
I understand your thought :)
Thank you for the detailed explanation.

Reply:
I'll follow up on this. I already toyed around a bit to get an idea how the current diagnostics functions could be changed to make them more customizable & composable.  (https://github.com/mfussenegger/dotfiles/compare/lsp-diag?expand=1#diff-e9e37b70bcd90d6df21d11736e98c583R19) 

My thoughts right now go into the direction of:

- Having the default publishDiagnostics callback as is
- Provide access to the diagnostics (bufnr -> linenr -> list of diagnostics)
- Make the logic that is currently invoked by the publishDiagnostics more composable, so users can replace the `publishDiagnostics` callback and easily re-use the parts they'd like to keep (e.g, store the diagnostics, active highlights)

Reply:
Rebased and added a second commit to drop the `message` handling.

I'll create a PR that adds something like 

```
local function diagnostics_to_items(bufnr, buf_diagnostics)
    local items = {}
    for linenr, diagnostics in pairs(buf_diagnostics) do
        if #diagnostics > 0 then
            local d = diagnostics[1]
            table.insert(items, {
                bufnr = bufnr,
                lnum = linenr + 1,
                vcol = 1,
                col = d.range.start.character + 1,
                text = d.message
            })
        end
    end
    return items
end
```

depending on the outcome of https://github.com/neovim/neovim/pull/11932